devito/ir/iet/utils.py

```diff
@@ -122,3 +123,34 @@ def derive_parameters(iet, drop_locals=False):
         parameters = [p for p in parameters if not (p.is_ArrayBasic or p.is_LocalObject)]
 
     return parameters
+
+
+def maybe_alias(obj, candidate):
+    """
+    True if `candidate` can act as an alias for `obj`, False otherwise.
+    """
+    if obj is candidate:
+        return True
+
+    # Names are unique throughout compilation, so this is another case we can handle
+    # straightforwardly. It might happen that we have an alias used in a subroutine
+    # with different type qualifiers (e.g., const vs not const, volatile vs not
+    # volatile), but if the names match, they definitely represent the same
+    # logical object
+    if obj.name == candidate.name:
+        return True
+
+    if obj.is_AbstractFunction:
```

Comment:
Are these case really needed? Will obviously always return false passed this line no matter

Reply:
not strictly speaking and not today, but they were in the back of my mind already, so I put them there for the future